dali/python/nvidia/dali/pipeline.py

```diff
@@ -1042,10 +1039,19 @@ def _next_op_id(self):
         return i
 
     def build(self):
-        """Build the pipeline.
+        """Build the pipeline (optional step).
 
-        Pipeline needs to be built in order to run it standalone.
-        Framework-specific plugins handle this step automatically.
+        Pipeline is automatically built when Pipeline is:
```

Comment:
I think that from the user's perspective it's more important to know what happens when the pipeline is built than explaining in detail when the automatic build may happen. I also think that parallel external source deserves a mention, because that's perhaps the single functionality that actually requires the manual invocation of `build`.

```suggestion
        Instantiates the pipeline's backend objects and starts processing threads. If the pipeline uses multi-processing ``external_source``, the worker processes are also started.

        Pipeline is automatically built when Pipeline is:
```

Reply:
I think we kept build for the rare case where it matters for the user to do as much preparation as possible without touching the data, that was my intention to keep it this way and mention when the pipeline will be otherwise built.

Reply:
Touching the data is only relevant for bencharks AFAIK - a more realistic use case is avoiding the creation of CUDA contexts when forking for parallel external source.